### DISCUSSION TITLE
Can you add an option in settings to make Windows 10 start menu look more modern?

### DISCUSSION THREAD

**Discussion by lixkote** [valinet/ExplorerPatcher#937] [Ideas]
Hello, i mean can we have an option to make the new 10 start menu to fit better with 11 theme?

**Comment by lixkote** [comment]
![image](https://user-images.githubusercontent.com/95425619/155852244-55b3eb15-bb8e-4ff5-8b29-305a68c252de.png)
Something like that

**Reply by valinet** [reply]
It’s not my implementation, it’s Microsoft’s, I just patched it to make it work again. I don’t know much about its internal implementation; if I figure something out, maybe, but it’s hard to have control over something that’s pretty obfuscated and that you do not have much control of.

**Reply by lixkote** [reply]
Btw: You can temporarily tweak the start menu in windows 10 through the debug options in visual studio and manual editing the xaml properties.
Further explanation: https://www.reddit.com/r/Windows10/comments/nt2r58/windows_11_start_menu_custom/
https://www.reddit.com/r/Windows10/comments/nvcrie/windows_11_start_menu_how_to_temporary_make_your/

**Reply by lixkote** [reply]
And if possible, try to replicate some functions from startallback so that the whole thing fits better with windows 11

**Reply by valinet** [reply]
> And if possible, try to replicate some functions from startallback so that the whole thing fits better with windows 11

Well, I actually prefer the Windows 10 aesthetic when it comes to the Windows shell, it’s one of the things I don’t like in StartAllBack, for example, the fact that it rounds the taskbar buttons.

> Btw: You can temporarily tweak the start menu in windows 10 through the debug options in visual studio and manual editing the xaml properties.

Well, that’s interesting. I heard about this before, mostly when people were hiding the Recommended section in the new menu with a similar method, there was a Reddit post about that. Someone also made a post on how to attach with the debugger and alter the contents of the function responsible for that in newer builds: https://www.reddit.com/r/Windows11/comments/ri3zu4/batch_file_to_remove_recommended_section_of_start/?utm_source=share&utm_medium=ios_app&utm_name=iossmf

Yeah, idk, I could give it a shot I guess. Do you know how to access that tree programmatically? I suppose it could be done by simply being in process, without attaching as a debugger. Thanks. Also thanks for the links.

P.S. Too bad the author there did not follow with a working implementation, since he already had the knowledge, it would’ve made the implementation a touch easier.

**Reply by valinet** [reply]
Anyway, the question of the day is how to access that visual tree thing programmatically, from a thread in the same process that we are interested in altering?

**Reply by valinet** [reply]
I also asked the author of those posts [here](https://www.reddit.com/user/henrik_z4/comments/s8lj62/i_finally_got_some_time_to_take_care_of/). I am really curious how to access that tree from code, it could indeed be really useful for a ton of stuff.

**Reply by valinet** [reply]
The author of the post did not want to share the top-secret details of their implementation that "will be open source"; anyway, I got this to work based on the guidance from the post, wrote an implementation for accessing and altering the live visual tree from scratch, available in the latest pre-release.

![image](https://user-images.githubusercontent.com/6503598/156178662-0d76911f-376f-45fc-9fc0-d4a34070eb54.png)

**Reply by lixkote** [reply]
Thanks <3

**Reply by chip33** [reply]
yes, great work Valinet !
